crates/ecs/src/systems/mod.rs

```diff
@@ -710,14 +738,67 @@ impl<'a, Component: Debug + Send + Sync + 'static + Sized> SystemParameter
     }
 }
 
-/*
+/// Segment for [`SystemParameter`] without data.
+#[derive(Debug, Default)]
+pub struct UnitSegment<T: Default> {
+    item: PhantomData<T>,
```

Comment:
What's the need for `T` when it's "without data", why not make it a unit struct?

Reply:
A segment needs to implement `IntoIterator<Item=Self>` where `Self` is the system parameter. `T` is the system parameter that the segment is for.